pypsa/linopt.py

```diff
@@ -601,6 +601,175 @@ def set_int_index(ser):
+        mkdir build
+        cd build
+        cmake ..
+        make
+        ctest
+
+    Then in .bashrc add paths of executables and library ::
+        export PATH="${PATH}:/foo/HiGHS/build/bin"
+        export LD_LIBRARY_PATH="${LD_LIBRARY_PATH}:/foo/HiGHS/build/lib"
+        source .bashrc
+
+    Now when typing ``highs`` in the terminal you should see something like ::
+        Running HiGHS 1.1.1 [date: 2021-11-14, git hash: 95342daa]
+
+    Architecture
+    -------------
+    The function reads and execute (i.e. subprocess.Popen,...) terminal
+    commands of the solver. Meaning the command can be also executed at your
+    command window/terminal if HiGHs is installed. Executing the commands on
+    your local terminal helps to identify the raw outputs that are useful for
+    developing the interface further.
+
+    All functions below the "process = ..." do only read and save the outputs
+    generated from the HiGHS solver. These parts are solver specific and
+    depends on the solver output.
+
+    Solver options
+    ---------------
+    Solver options are read by the 1) command window and the 2) option_file.txt
+
+    1) An example list of solver options executable by the command window is given here:
+    Examples:
+    --model_file arg 	File of model to solve.
+    --presolve arg 	    Presolve: "choose" by default - "on"/"off" are alternatives.
+    --solver arg 	    Solver: "choose" by default - "simplex"/"ipm" are alternatives.
+    --parallel arg 	    Parallel solve: "choose" by default - "on"/"off" are alternatives.
+    --time_limit arg 	Run time limit (double).
+    --options_file arg 	File containing HiGHS options.
+    -h, --help 	        Print help.
+
+    2) The options_file.txt gives some more options, see a full list here: 
+    https://www.maths.ed.ac.uk/hall/HiGHS/HighsOptions.set 
+    By default, we insert a couple of options for the ipm solver. The dictionary
+    can be overwritten by simply giving the new values. For instance in PyPSA-Eur,
+    you could write a dictionary replacing some of the default values or adding new
+    options:
+    ```
+    solving:
+        solver:
+            name: highs,
+            method: ipm,
+            parallel: "on",
+            <option_name>: <value>,
+    ```
+    Note, the <option_name> and <value> must be equivalent to the name convention
+    of HiGHS. Some function exist that are not documented, check their GitHub file:
+    \HighsOptions.h
+
+    Output
+    ------
+    status : string,
+        "ok" or "warning"
+    termination_condition : string,
+        Contains "optimal", "infeasible", 
+    variables_sol : series
+    constraints_dual : series
+    objective : float
+    """
+    options_fn = "highs_options.txt"
+    default_dict = {
+        "method": "ipm",
+        "primal_feasibility_tolerance": 1e-04,
+        "dual_feasibility_tolerance": 1e-05,
+        "ipm_optimality_tolerance": 1e-6,
+        # "ipm_iteration_limit": 100,
```

Comment:
Remove if commented out.

Reply:
done in https://github.com/PyPSA/PyPSA/pull/308/commits/9becfe64b6a12c6d1f9640d923b431192b2073d4